### PR TITLE
helix: fix hx.exe installation dir

### LINKED ISSUE #22026: Helix editor in lib folder
### Description / Steps to reproduce the issue

When installing [Helix Editor](https://packages.msys2.org/packages/mingw-w64-ucrt-x86_64-helix) via pacman, the 'hx' executable is being placed in '/ucrt64/lib/helix/hx.exe' instead of '/ucrt64/bin' (if instaling into the ucrt env)

There is a shell script named 'helix' ( without .sh) in the bin directory, but this only gets executed inside the msys2 ucrt shell.

### Expected behavior

All (at least most) other commandline executables in 'bin' can be run from a 'cmd' shell or Powershell (if the environment path variable is correcty set to these paths). Anyway, the standard command to run helix on other systems (Linux, Macos) is hx, not helix, and hx does not even work inside the 'ucrt' environment.

### Actual behavior

When having a correctly setup msys2 with 'path' pointing to the standard bin directories, helix cannot be invoked by hx in any shell. It can however be invoked from inside the ucrt bash with the 'helix' shell script. But the hx command should work too, and should work from 'cmd' and 'powershell'.

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MSYS_NT-10.0-19045

### PR DESCRIPTION
**Pull request by raedrizqie**
Fixes #22026

### CODE REVIEW

File: mingw-w64-helix/PKGBUILD [hunk 1]
```diff
@@ -57,52 +55,52 @@ END
 }
 
 build() {
-  cd "${srcdir}"
-  rm -rf "build-${MSYSTEM}" | true
+  rm -rf "build-${MSYSTEM}"
   cp -r "${_realname}-${pkgver}" "build-${MSYSTEM}"
   cd "build-${MSYSTEM}"
 
+  # FIXME: ${MINGW_PREFIX} will expand to CI installation prefix
```
Comment (ognevny) [inline mingw-w64-helix/PKGBUILD:62]: `MSYS2_ENV_CONV_EXCL?`

Comment (raedrizqie) [reply mingw-w64-helix/PKGBUILD:62]: tried it, but didnt work :(

Comment (ognevny) [reply mingw-w64-helix/PKGBUILD:62]: there is a [code](https://github.com/msys2/MINGW-packages/issues/22026#issuecomment-2373264301) that defines default path. can it be patched?

**Comment by ognevny** [comment]
you said it doesn't work, but you pushed that change?..

**Comment by raedrizqie** [comment]
> MSYS2_ARG_CONV_EXCL

previously, i tried `MSYS2_ARG_CONV_EXCL="HELIX_DEFAULT_RUNTIME="`

**Comment by ognevny** [comment]
> > MSYS2_ARG_CONV_EXCL
> 
> previously, i tried `MSYS2_ARG_CONV_EXCL="HELIX_DEFAULT_RUNTIME="`

I was talking about another one: `MSYS2_ENV_CONV_EXCL`, not `...ARG...` one

**Comment by raedrizqie** [comment]
> > > MSYS2_ARG_CONV_EXCL
> > 
> > 
> > previously, i tried `MSYS2_ARG_CONV_EXCL="HELIX_DEFAULT_RUNTIME="`
> 
> I was talking about another one: `MSYS2_ENV_CONV_EXCL`, not `...ARG...` one

ah.. my bad

File: mingw-w64-helix/PKGBUILD [hunk 2]
```diff
@@ -57,52 +55,52 @@ END
...
     install -Dm 755 {} -t "$runtime_dir/grammars" \;
   install -Dm644 runtime/tutor -t "$runtime_dir"
-  cp -a "${runtime_dir}/" "${pkgdir}${MINGW_PREFIX}/lib/${_realname}/runtime/"
 
-  install -Dm644 "contrib/completion/hx.bash" "${pkgdir}${MINGW_PREFIX}/share/bash-completion/completions/${_realname}"
-  install -Dm644 "contrib/completion/hx.fish" "${pkgdir}${MINGW_PREFIX}/share/fish/vendor_completions.d/${_realname}.fish"
-  install -Dm644 "contrib/completion/hx.zsh" "${pkgdir}${MINGW_PREFIX}/share/zsh/site-functions/_${_pkgname}"
+  install -Dm644 "contrib/completion/hx.bash" "${pkgdir}${MINGW_PREFIX}/share/bash-completion/completions/hx"
+  install -Dm644 "contrib/completion/hx.fish" "${pkgdir}${MINGW_PREFIX}/share/fish/vendor_completions.d/hx.fish"
+  install -Dm644 "contrib/completion/hx.zsh"  "${pkgdir}${MINGW_PREFIX}/share/zsh/site-functions/_hx"
```
Comment (ognevny) [inline mingw-w64-helix/PKGBUILD:100]: formatting?

**Comment by ognevny** [comment]
it now works and bash script can be removed as executable launches even with double-click on file

File: mingw-w64-helix/PKGBUILD [hunk 3]
```diff
@@ -57,52 +55,52 @@ END
 }
 
 build() {
-  cd "${srcdir}"
-  rm -rf "build-${MSYSTEM}" | true
+  rm -rf "build-${MSYSTEM}"
   cp -r "${_realname}-${pkgver}" "build-${MSYSTEM}"
   cd "build-${MSYSTEM}"
 
+  export MSYS2_ENV_CONV_EXCL="*"
```
Comment (ognevny) [inline mingw-w64-helix/PKGBUILD:62]: better use
```suggestion
  export MSYS2_ENV_CONV_EXCL='HELIX_DEFAULT_RUNTIME'
```

Comment (raedrizqie) [reply mingw-w64-helix/PKGBUILD:62]: this still doesnt work.. try running `hx --health`
we need to patch `lib.rs`

Comment (ognevny) [reply mingw-w64-helix/PKGBUILD:62]: you can try to change [this line](https://github.com/helix-editor/helix/blob/b18a471ed189fb326a781181a28f3073f5c1fe1e/helix-loader/src/lib.rs#L42)

**Comment by ognevny** [comment]
thanks!

**Pull request merged**